package caffe;

// message Image {
//   message Pixel {
//     required int32 x = 1;
//     required int32 y = 2;
//     required double value = 3;
//   }
//   required int32 width = 1;
//   required int32 height = 2;
//   repeated Pixel pixels = 3;
// }

message Multi {
  required int32 width = 1;
  required int32 height = 2;
  required int32 num_channels = 3;

  // message Pixel {
  //   required int32 x = 1;
  //   required int32 y = 2;
  //   required int32 channel = 2;
  //   required double value = 4;
  // }
  // repeated Pixel pixels = 4;

  // repeated int32 x = 5;
  // repeated int32 y = 6;
  // repeated int32 channel = 7;
  // repeated double value = 8;

  required int32 x_stride = 9;
  required int32 y_stride = 10;
  required int32 channel_stride = 11;
  repeated double elem = 12;
}
